### ISSUE TITLE
Other alternative in comparison

### ISSUE THREAD

**Issue by wyfo** [ltworf/typedload#173]
Could it be possible to mention [apischema](https://wyfo.github.io/apischema/) as an alternative in the comparison section of the documentation?

**Comment by ltworf** [comment]
I don't see a link nor mention to typedload in that website, even in the benchmark section.

in `docs/comparisons.md` there are comparisons with other similar libraries.

Are you willing to make a PR there?

**Comment by wyfo** [comment]
> I don't see a link nor mention to typedload in that website, even in the benchmark section.

In fact, i've discover your library only recently, and the benchmark section is tied to *pydantic*'s one for the moment (it's a fork, bias included, with the only addition of *apischema* implementation).
For the moment, it has no comparison section, except a very detailed one about *pydantic*. A more general section isn't currently planned, and if the mention of *apischema* in *typedload* documentation is tied to its mention in the other side, I understand and quite agree with that.

> in `docs/comparisons.md` there are comparisons with other similar libraries.
>
> Are you willing to make a PR there?

Being the author of the mentioned library, I don't think to be the right guy to do that, that's why I used an issue and not a PR, but I could, depending on the first part of my response.

**Comment by ltworf** [comment]
It would be good if your examples were fully functioning code I could just copy and paste.

**Comment by wyfo** [comment]
Yet, every of the 100 examples present in the documentation are normally valid and executing properly; they can be found in the the `examples` directory of the repository (and they are included as-is in the documentation with markdown extension). As stated in the documentation:
> This example and further ones are using pytest stuff because they are in fact run as tests in the library CI

Even if they are automatically wrapped to be executed by *pytest*, they still work as-is.
I've just executed the following lines to prove it:
```bash
python3.9 -m venv venv # example are written using latest python release
venv/bin/pip install apischema
venv/bin/pip install pytest # pytest.raised is used in the example, as they are wrapped in the CI
venv/bin/pip install graphql-core  # this dependency is necessary when executing graphql examples
venv/bin/pip install sqlalchemy pydantic attrs # these additional dependencies are required for complex examples of the document Examples section
find examples/ -name "*.py" -exec venv/bin/python3.9 {} \;  # you can replace examples/ by your path to the example folder of apischema repository
```
And every test passes without error (two tests will log a stack trace, but t's part of the tests); at least, that's true on macOS (and it should be the case on Ubuntu as the CI is run on it), but I admit I haven't tested on Debian.

That's being said, I assume the documentation is not clear enough about the requirements to run the examples, I will solve this issue thanks to you.

Moreover, if some of the dependencies are optional to execute most of the examples, the documentation has been wanting to be up-to-date with the latest Python release, so Python 3.9 specificities (especially PEP 585, as it is noticeable from the *README* example) are used. However, all tests are executed with the four minor Python versions from 3.6 to 3.9, but also with *pypy3*. To make this possible without modifying the examples code, all tests are prefixed (automatically in the CI) with the content of `scripts/test_wrapper.py`. You can use this prefix to execute the examples in an old Python version.

If, despite these precisions, you're still encountering issue in running the examples, I would be grateful for giving me some precisions about it, if you are interested in its resolution.

**Comment by wyfo** [comment]
With the last release of *apischema*, you can now use `pip install apischema[examples]` and execute examples as explained [here](https://wyfo.github.io/apischema/#run-the-documentation-examples).

**Comment by ltworf** [comment]
Seems that @wyfo did not do the requested changes to his website.

**Comment by wyfo** [comment]
> Seems that @wyfo did not do the requested changes to his website.

Indeed, I did not wanted to add other libraries to benchmark before having my own benchmark.
But now, *apischema* has its own benchmark, and *typedload* has been [added to it](https://wyfo.github.io/apischema/0.17/optimizations_and_benchmark/)!

**Comment by ltworf** [comment]
Well, seems I can't escape now :)

Meanwhile now there is an actual performance testsuite so it shouldn't be as hard to implement… and I should maybe publish the results on the website rather than relying on interested people to run it themselves.

**Comment by ltworf** [comment]
Mention added in a0792d7e9f00ef60fd714e9102a911ae4a1de913
